Sort items in people view

AlexPerathoner opened the pull request:
Relates to  #619
### Changes made
Added sort dropdown to sort items in `/person/[id]` views by Vote count (former method), oldest / newest release date

![immagine](https://github.com/user-attachments/assets/0c210e0b-6599-4fc5-9da6-93add6eba10b)

AlexPerathoner committed:
Sort items in people view

IRHM commented on src/routes/(app)/person/[id]/+page.svelte:
```diff
@@ -19,6 +19,9 @@
   let pageError: Error | undefined;
+  let sortOption = "Vote count";
+  let cachedCredits: TMDBPersonCombinedCredits | undefined;
+  let sortedCredits: TMDBPersonCombinedCredits | undefined;
```
Instead of having two lists, we could just have one `credits` array and sort it in place whenever a the sort is changed?

IRHM commented on src/routes/(app)/person/[id]/+page.svelte:
```diff
@@ -31,33 +34,64 @@
-    })();
+      person = await getPerson(personId);
+      cachedCredits = await getPersonCredits();
+      sortedCredits = getSortedCredits(cachedCredits, sortOption);
```
If we just have the one array of credits, `getSortedCredits` could become `sortCredits`, which would simply sort the one credits array (then also wouldn't have to return anything)

IRHM commented on src/routes/(app)/person/[id]/+page.svelte:
```diff
@@ -31,33 +34,64 @@
+
+  function getSortedCredits(credits: TMDBPersonCombinedCredits, sortOption: string) {
+    if (!credits || !credits.cast) return;
+    let sorted = [...credits.cast];
```
with the one array, we wouldn't need to spread the credits into this intermediary array to sort (which I believe can be expensive), we can just run `credits.sort` below (i believe that should work)

IRHM commented on src/routes/(app)/person/[id]/+page.svelte:
```diff
@@ -31,33 +34,64 @@
+      case "Vote count":
+        sorted.sort((a, b) => b.vote_count - a.vote_count);
+        break;
+      case "Newest":
```
The Newest and Oldest filters seem to be able to break on certain credits lists.

It works on your example page, but for example, can break on [Uma Thurman (/person/139)](http://localhost:5173/person/139).

With her list as an example, if I set the sort to Newest, the list looks right, until I get towards the bottom, where it seems to have just stopped sorting (same thing happens, but opposite with Oldest sort).

It seems to always break after the movie "The Old Guard 2" (in her list specifically), which I don't think has a release date yet, which _might_ be why the sort breaks.

IRHM commented on src/routes/(app)/person/[id]/+page.svelte:
```diff
@@ -31,33 +34,64 @@
+        );
+        break;
+    }
+    return { ...credits, cast: sorted };
```
with the one array, i believe we shouldn't have to return anything - though possibly we may need to reassign credits = credits to get it to update ui (not 100% certain on that)

IRHM reviewed (changes requested):
Hi @AlexPerathoner, thank you for taking the time to do this!

I noticed one bug with the sorting and one small thing I think we could improve upon (we could possibly be using just one list instead of two for storing the credits).

Let me know if I can help with anything (or if you want me to finish any of this)!

AlexPerathoner commented:
Removed the two cached arrays into just one; updated methods to reflect that; updated sort method to treat shows which don't have a release date nor a first air date as unreleased (so they appear first when sorting by Newest and last when by Oldest).

Also noticed that some people have duplicate items in them (because of different `character` and `credits_id` (for example https://api.themoviedb.org/3/person/25072/combined_credits has 59941 twice, once as `Self` and once as `Self - Guest`). Since in the people view there's no reference to the played character yet, I added a filter to remove these duplicates.

IRHM committed:
person: Add max-width to .filters ctr, remove min-width, combine with .container

- Removed min-width so improve responsivity for mobile (overflows).
- Removed unused styles (margin-top/bottom that were overridden in .container).
- Add max-width to match PosterList.
- Remove .container and add styles to .filters (combined them).

IRHM committed:
DropDown: showActiveElementsInOptions: Just use `options` when true

No need for the filter when this is true, we can just pass the options var.

IRHM committed:
person: sortCredits: Remove use of `sorted` intermediary array, sort credits array directly

I believe this should be more performant (removing the use of spread operator) and is just a little cleaner.

IRHM reviewed (approved):
This is awesome, thank you!

I'm liking the less duplicates in the person view, it has always annoyed me!

p.s. I made a few tiny changes, didn't find any bugs, just changed around some code/styling.
